hypershift-operator/controllers/hostedcluster/internal/platform/powervs/powervs.go

```diff
@@ -0,0 +1,291 @@
+	"github.com/openshift/hypershift/support/upsert"
+)
+
+const (
+	// TODO(mkumatag): Move to OpenShift built image
+	imageCAPIBM = "k8s.gcr.io/capi-ibmcloud/cluster-api-ibmcloud-controller:v0.2.0"
+)
+
+type PowerVS struct {
+}
+
+func (p PowerVS) DeleteCredentials(ctx context.Context, c client.Client, hcluster *hyperv1.HostedCluster, controlPlaneNamespace string) error {
+	//TODO(mkumatag): implement me
+	return nil
+}
+
+func (p PowerVS) ReconcileCAPIInfraCR(ctx context.Context, c client.Client, createOrUpdate upsert.CreateOrUpdateFN,
+	hcluster *hyperv1.HostedCluster,
+	controlPlaneNamespace string,
+	apiEndpoint hyperv1.APIEndpoint) (client.Object, error) {
+	ibmCluster := &capiibmv1.IBMPowerVSCluster{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: controlPlaneNamespace,
+			Name:      hcluster.Name,
+		},
+	}
+
+	ibmCluster.Spec.ServiceInstanceID = hcluster.Spec.Platform.PowerVS.ServiceInstanceID
+
+	_, err := createOrUpdate(ctx, c, ibmCluster, func() error {
+		ibmCluster.Annotations = map[string]string{
+			capiv1.ManagedByAnnotation: "external",
+		}
+
+		// Set the values for upper level controller
+		ibmCluster.Status.Ready = true
+		ibmCluster.Spec.ControlPlaneEndpoint = capiv1.APIEndpoint{
+			Host: apiEndpoint.Host,
+			Port: apiEndpoint.Port,
+		}
+		return nil
+	})
+	if err != nil {
+		return nil, err
+	}
+	// reconciliation strips TypeMeta. We repopulate the static values since they are necessary for
+	// downstream reconciliation of the CAPI Cluster resource.
+	ibmCluster.TypeMeta = metav1.TypeMeta{
+		Kind:       "IBMPowerVSCluster",
+		APIVersion: capiibmv1.GroupVersion.String(),
+	}
+	return ibmCluster, nil
+}
+
+func (p PowerVS) CAPIProviderDeploymentSpec(hcluster *hyperv1.HostedCluster, _ *hyperv1.HostedControlPlane) (*appsv1.DeploymentSpec, error) {
+	defaultMode := int32(420)
+	deploymentSpec := &appsv1.DeploymentSpec{
+		Template: corev1.PodTemplateSpec{
+			Spec: corev1.PodSpec{
+				TerminationGracePeriodSeconds: k8sutilspointer.Int64Ptr(10),
+				Tolerations: []corev1.Toleration{
+					{
+						Key:    "node-role.kubernetes.io/master",
+						Effect: corev1.TaintEffectNoSchedule,
+					},
+				},
+				Volumes: []corev1.Volume{
+					{
+						Name: "capi-webhooks-tls",
+						VolumeSource: corev1.VolumeSource{
+							Secret: &corev1.SecretVolumeSource{
+								DefaultMode: &defaultMode,
+								SecretName:  "capi-webhooks-tls",
+							},
+						},
+					},
+					{
+						Name: "credentials",
+						VolumeSource: corev1.VolumeSource{
+							Secret: &corev1.SecretVolumeSource{
+								SecretName: hcluster.Spec.Platform.PowerVS.NodePoolManagementCreds.Name,
+							},
+						},
+					},
+					{
+						Name: "svc-kubeconfig",
+						VolumeSource: corev1.VolumeSource{
+							Secret: &corev1.SecretVolumeSource{
+								DefaultMode: &defaultMode,
+								SecretName:  "service-network-admin-kubeconfig",
+							},
+						},
+					},
+					{
+						Name: "token",
```

Comment:
why is this volume needed?

Reply:
ah! copy paste left over, not needed. Will remove.